### ISSUE TITLE
Error after upgrade to 1.19.3 : services.web.depends_on contains an invalid type, it should be an array (old docker-compose v1 and use-docker-compose-from-path)

### ISSUE THREAD

**Issue by urashima82** [drud/ddev#3894]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Run a Diagnostic and Paste Link Here

https://gist.github.com/urashima82/9648781d9fbe79fd6f1bfc5ec855f78c

### Current Behavior

After upgrading ddev from 1.19.2 to 1.19.3 when I start a project I have this error :
```
The Compose file '/PATH_TO_PROJECT/.ddev/.ddev-docker-compose-base.yaml' is invalid because: 
services.web.depends_on contains an invalid type, it should be an array 
Failed to start my-project: ComposeCmd failed to run 'COMPOSE_PROJECT_NAME=ddev-my-project docker-compose -f /PATH_TO_PROJECT/.ddev/.ddev-docker-compose-base.yaml config', action='[config]', err='exit status 1', stdout='', stderr='The Compose file '/PATH_TO_PROJECT/.ddev/.ddev-docker-compose-base.yaml' is invalid because:
services.web.depends_on contains an invalid type, it should be an array' 

```

I am running DDEV on Ubuntu, and installed using Homebrew.

### Expected Behavior

It simply should work.

### Anything else?

To fix my problem I had to revert to ddev 1.19.2

Here is the process to **downgrade** ddev to specific version : 1.19.2

1. Create a folder that will contains Formula for `brew install` command: `mkdir ddev-1.9.2`
2. Go inside the newly created directory : `cd ddev-1.9.2`
3. Download the `.rb` file in the `Formula` folder from https://github.com/drud/homebrew-ddev using file history to find the version you desire
4. Display the raw file and use the URL to download it :
`wget https://raw.githubusercontent.com/drud/homebrew-ddev/f6a8174be50e5a5e1d3edb25fe1815842743e5ce/Formula/ddev.rb`
5. Unlink current version of ddev from brew : `brew unlink ddev`
6. Install the version of ddev you have downloaded : `brew install ./ddev.rb`

**Comment by rfay** [comment]
Hi @urashima82 - You were requested to run the diagnostic, which would have given me lots of information. I can guess from your instructions that you are on Linux or macOS, but. that's about all I can guess. Could you please run the diagnostic `ddev debug test` and put it in a gist on gist.github.com as requested when creating the issue?

The FAQ explains how to install specific versions of ddev; homebrew is not good at that any more. See "How can I install a specific version of DDEV", https://ddev.readthedocs.io/en/latest/users/faq/

Did you do a `ddev poweroff` ?

**Comment by urashima82** [comment]
Hi @rfay,
Thanks for your answer, I've just add the result of de the diagnostic you can see in the original comment.
Of course I did a `ddev poweroff` before doing upgrade.

**Comment by rfay** [comment]
Thanks - You have the global ddev setting ` use-docker-compose-from-path=true` and you have a very old docker-compose v1 on your machine. `ddev config -global -use-docker-compose-from-path=false` and I think it will go better. I imagine that if you had the last docker-compose v1 (1.29) it would also work, but I'm not sure. Anyway, please turn that off and see.

**Comment by urashima82** [comment]
Thank you ! You were right !!!
I did this command : `ddev config global -use-docker-compose-from-path=false` (without the dash before `global`)
It perfectly works now :)

**Comment by urashima82** [comment]
I was trying to make understand PHPStorm which PHP version is used (there is a bug with php8). I've tried to use the docker-compose installed through apt but it was definitely a mistake ;)

**Comment by rfay** [comment]
Yes, that's annoying. The instructions say how to get past that though, see https://ddev.readthedocs.io/en/latest/users/topics/phpstorm/

But PhpStorm says that bug is fixed in upcoming 2022.2, yay!
